krm-functions/scripts/unit-test.sh

```diff
@@ -0,0 +1,31 @@
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+err=0
+for d in */ ; do
+  if [ $d != "scripts/" ]; then
+    for e in "$d"*/ ; do (
+      echo running unit tests for "$e" ...
+      cd "$e" || exit
+      go test -cover -v ./... || exit
```

Comment:
Is there a problem if we run the go tests in the parent directory instead of `cd` to each subdirectory?

Reply:
I get the following error:

```
go: cannot find main module, but found .git/config in {PATH}/sigs.k8s.io/krm-functions-registry
        to create a module there, run:
        cd .. && go mod init
```

Reply:
Got it